spirv_cross_containers.hpp

```diff
@@ -219,7 +220,13 @@ class SmallVector : public VectorView<T>
 		this->buffer_size = count;
 	}
 
-	SmallVector(std::initializer_list<T> init) SPIRV_CROSS_NOEXCEPT : SmallVector(init.begin(), init.end())
+	template <typename U>
+	SmallVector(std::initializer_list<U> init) SPIRV_CROSS_NOEXCEPT : SmallVector(init.begin(), init.end())
+	{
+	}
+
+	template <typename U, size_t M>
+	SmallVector(U (&init)[M]) SPIRV_CROSS_NOEXCEPT : SmallVector(init, &init[M])
```

Comment:
`const U`?

Reply:
`&init[M]` is not a valid dereference. I've had problems where Clang inferred things from just this before. `init + M` seems better here to avoid that potential problem.

Reply:
> `const U`?

I was afraid that wouldn't compile. Apparently, it does.

> `&init[M]` is not a valid dereference. I've had problems where Clang inferred things from just this before. `init + M` seems better here to avoid that potential problem.

I was actually torn between these two forms. Ironically, I avoided the latter because I was afraid *that* was undefined behavior! But I think you're right and I got that backwards.